internal/collectors/lsmod.go

```diff
@@ -0,0 +1,83 @@
+package collectors
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"sort"
+	"strings"
+
+	"github.com/SUSE/connect-ng/internal/util"
+)
+
+type LSMOD struct {
+	UpdateDataIDs bool
+}
+
+const KernModChecksumFile = "chksum-kernmoddata.txt"
+const lsmodTag = "mod_list"
+
+func (lsmod LSMOD) run(arch string) (Result, error) {
+	util.Debug.Print("lsmod.UpdateDataIDs: ", lsmod.UpdateDataIDs)
+	modInfo, err := util.Execute([]string{"lsmod"}, nil)
+	// Should not happen, but if command fails
+	// send nil to SCC to indicate that module data was unavailable.
+	if err != nil {
+		return Result{lsmodTag: nil}, err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(modInfo))
+
+	// Iterate through each line get module name
+	modList := make(map[string]bool)
+	for scanner.Scan() {
+		module := strings.Fields(scanner.Text())[0]
+		// skip "Mudule" from the hearder line
+		if module != "Module" {
```

Comment:
Instead of this, call scanner.Scan before the for loop so the first line is skipped (which is where the header appears).

Reply:
great idea, thanks